noir-projects/noir-protocol-circuits/crates/types/src/address/aztec_address.nr

```diff
@@ -52,16 +66,55 @@ impl Deserialize<AZTEC_ADDRESS_LENGTH> for AztecAddress {
     }
 }
 
+impl ToPoint for AztecAddress {
+    fn to_point(self) -> Point {
+        // Calculate y^2 = x^3 - 17
+        let y_squared = pow(self.inner, 3) - 17;
+
+        // TODO (#8970): Failure validation
+        // We can see if y is square first, or we can soft fail with just sqrt(y_squared);
+        // If y is not square, the x-coordinate is not on the curve
+        // Do we throw here or can we soft continue and this is fine ? Test this with points not on curve to see what happens.
+        // let y_is_square = is_square(y_squared);
+        // assert(y_is_square);
+        let mut y = sqrt(y_squared);
+
+        // If we get a negative y coordinate, we pin it to the positive one by subtracting it from the Field modulus
+        if (!(y.lt(BN254_FR_MODULUS_DIV_2) | y.eq(BN254_FR_MODULUS_DIV_2))) {
+            y = (BN254_FR_MODULUS_DIV_2 + BN254_FR_MODULUS_DIV_2 + 1) - y;
+        }
+
+        Point { x: self.inner, y, is_infinite: false }
+    }
+}
+
 impl AztecAddress {
     pub fn zero() -> Self {
         Self { inner: 0 }
     }
 
-    pub fn compute(pub_keys_hash: PublicKeysHash, partial_address: PartialAddress) -> AztecAddress {
-        AztecAddress::from_field(poseidon2_hash_with_separator(
+    pub fn compute_preaddress(
+        pub_keys_hash: PublicKeysHash,
+        partial_address: PartialAddress,
+    ) -> Field {
+        poseidon2_hash_with_separator(
             [pub_keys_hash.to_field(), partial_address.to_field()],
             GENERATOR_INDEX__CONTRACT_ADDRESS_V1,
-        ))
+        )
+    }
+
+    pub fn compute(public_keys: PublicKeys, partial_address: PartialAddress) -> AztecAddress {
+        let public_keys_hash = public_keys.hash();
+
+        let pre_address = poseidon2_hash_with_separator(
+            [public_keys_hash.to_field(), partial_address.to_field()],
+            GENERATOR_INDEX__CONTRACT_ADDRESS_V1,
+        );
+
+        let address_point = derive_public_key(EmbeddedCurveScalar::from_field(pre_address)).add(
+            public_keys.ivpk_m.to_point(),
+        );
+        AztecAddress::from_field(address_point.x)
```

Comment:
From the comments in https://github.com/AztecProtocol/aztec-packages/issues/8966 it seemed like we needed to do `addr = -addr` if the y coordinate was high - does that not affect the x coord?

Reply:
I see, I thought maybe the x-coord would be affected. I'd maybe consider however making a point as to how multiple actors will later try to recover the point from the x coord, and if the original y coord was high, they'll recover a _different_ point - but this is fine, because they always recover the same one.